pyedflib/edfwriter.py

```diff
@@ -136,29 +142,28 @@ def check_signal_header_correct(channels, i, file_type):
                                                                       str(ch['physical_max'])[:8]))
 
 
-
-def u(x):
+def u(x: bytes) -> str:
     return x.decode("utf_8", "strict")
 
 
-def du(x):
+def du(x: Union[str, bytes]) -> bytes:
     if isinstance(x, bytes):
         return x
     else:
         return x.encode("utf_8")
 
 
-def isstr(s):
+def isstr(s: Any) -> bool:
     warnings.warn("Function 'isstr' is deprecated.", DeprecationWarning, stacklevel=2)
     return isinstance(s, str)
 
 
-def isbytestr(s):
+def isbytestr(s: Any) -> bool:
     warnings.warn("Function 'isbytestr' is deprecated.", DeprecationWarning, stacklevel=2)
     return isinstance(s, bytes)
 
 
-def sex2int(sex):
+def sex2int(sex: Union[int, str, None]) -> Optional[int]:
```

Comment:
Technically it always returns an `int` or `None`, but can raise an Exception, would that still be `Optional` then?

```suggestion
def sex2int(sex: Union[int, str, None]) -> Optional[int, None]:
```

Reply:
As specified in [PEP 484](https://peps.python.org/pep-0484/#exceptions), exceptions are out of scope of type hints. I guess it won't be required to add None as inside the Optional neither.

Reply:
Yep indeed didn't know that! Thanks :)